### DISCUSSION TITLE
Safe to import types from api routes?

### DISCUSSION THREAD

**Discussion by KATT** [vercel/next.js#20462] [Help]
Hey!

I generally like type inference more than pre-defined types and wondering if it's generally safe to export types from API routes and use them in my components?

I've made an example here: https://github.com/KATT/nextjs-with-typescript-and-type-inference

In this example, I've created an API-route from which I'm inferring types into my pages & components. Is it safe to do this? Feels bad to import `api/`-stuff into React-components.

- Example **deployment**: https://nextjs-with-typescript-and-type-inference.vercel.app/users
- **Exporting** type `GetUsersResponse` in [`api/users/index.tsx`](https://github.com/KATT/nextjs-with-typescript-and-type-inference/blob/a8b50ec3cc57aff677819737a1901d344ebfb513/pages/api/users/index.ts#L13)
- **Using** said type in 
  - [`pages/users/index.tsx`](https://github.com/KATT/nextjs-with-typescript-and-type-inference/blob/a8b50ec3cc57aff677819737a1901d344ebfb513/pages/users/index.tsx#L26)
  - [`List`](https://github.com/KATT/nextjs-with-typescript-and-type-inference/blob/a8b50ec3cc57aff677819737a1901d344ebfb513/components/List.tsx#L6) / [`ListItem`](https://github.com/KATT/nextjs-with-typescript-and-type-inference/blob/main/components/ListItem.tsx#L4)

<img width="607" alt="Screenshot 2020-12-24 at 22 22 52" src="https://user-images.githubusercontent.com/459267/103106531-9a8c5b00-4636-11eb-936e-9c8d282628ec.png">

**Comment by KATT** [comment]
Oh, I just saw this tool linked in the docs! https://next-code-elimination.now.sh

Think that answers my question :)

**Reply by mauroc8** [reply]
So basically the answer is that it's not safe?
